src/NullCache.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Fig\SimpleCache;
+
+use Psr\SimpleCache\CacheInterface;
+
+/**
+ * This cache can be used to avoid conditional cache calls.
+ */
+class NullCache implements CacheInterface
+{
+    /**
+     * {inheritdoc}
+     */
+    public function get($key, $default = null)
+    {
+        return $default;
+    }
+
+    /**
+     * {inheritdoc}
+     */
+    public function getMultiple($keys, $default = null)
```

Comment:
You think ´return array_fill_keys($keys, $default);` would be more suitable here.

Reply:
Done.

Reply:
True, though it seems weird it would ever be implemented as an iterator; `iterator_to_array` works with Traversable. `array_fill_keys(iterator_to_array($keys), $default)`

Reply:
True, though it seems weird it would ever be implemented as an iterator; `iterator_to_array` works with Traversable. `array_fill_keys(iterator_to_array($keys), $default)`

Reply:
True, though it seems weird it would ever be implemented as an iterator; `iterator_to_array` works with Traversable. `array_fill_keys(iterator_to_array($keys), $default)`

Reply:
True, though it seems weird it would ever be implemented as an iterator; `iterator_to_array` works with Traversable. `array_fill_keys(iterator_to_array($keys), $default)`

Reply:
`$keys` could be an `Iterable` or a `Traversable`.
Then in this case, the `foreach` construction is the best suitable choice here.

Reply:
I'm not sure what you mean, in which case wouldn't `iterator_to_array()` work?

Reply:
`$keys` can be an array or an object (which is `Traversable` or `Iterable`), and then, therefore, the foreach construction is the best choice here.

Reply:
Oh, right, for some reason I thought `iterator_to_array()` took `array` as well.
Well, I guess it's about preference at this point but you could do this: https://3v4l.org/6iIpK

Reply:
Oooh I see! Thanks for the example, it's always easier to understand :-)
I fully agree with it, but now, would it be better to use it or use a simple foreach ?
What would be the best in this case ?
I think, performance wise, the foreach would be faster.

Reply:
Tried both ways and seem similar in speed, `foreach` uses less optcode steps.
But it isn't really enough difference to care about imo, so I'd say it's just about preference. `foreach` seem simpler when you have to do ternary.

Reply:
I'm the first to use pure php functions, I like that, usually it increases maintainability. However, in this case, the foreach is definitely better I think.